xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-api/src/main/java/org/xwiki/notifications/internal/email/AbstractMimeMessageIterator.java

```diff
@@ -151,9 +139,7 @@ private void computeNext()
             try {
                 // TODO: in a next version, it will be import to paginate these results and to send several emails
                 // if there is too much content
-                this.currentEvents = notificationManager.getEvents(serializer.serialize(currentUser),
-                        NotificationFormat.EMAIL, false, Integer.MAX_VALUE / 4, null,
-                        lastTrigger, Collections.emptyList());
+                this.currentEvents = retrieveCompositeEventList(currentUser);
```

Comment:
At some point we need to handle the "paginate" results. IE: if there is too many events, send several emails instead of only one. I think `retrieveCompositeEventList` should return either or not some events needs to be retrieve afterwards...